toolboxes/feel/feelmodels/maxwell/maxwell.cpp

```diff
@@ -0,0 +1,551 @@
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+/**
+   \file maxwell.cpp
+   \author Romain Hild <romain.hild@unistra.fr>
+   \date 2018-05-03
+ */
+
+#include <feel/feelmodels/maxwell/maxwell.hpp>
+
+#include <feel/feelvf/vf.hpp>
+#include <feel/feelmodels/modelmesh/createmesh.hpp>
+
+namespace Feel
+{
+namespace FeelModels
+{
+
+MAXWELL_CLASS_TEMPLATE_DECLARATIONS
+MAXWELL_CLASS_TEMPLATE_TYPE::Maxwell( std::string const& prefix,
+                                      bool buildMesh,
+                                      WorldComm const& worldComm,
+                                      std::string const& subPrefix,
+                                      ModelBaseRepository const& modelRep )
+    :
+    super_type( prefix, worldComm, subPrefix, modelRep ),
+    M_epsilon( doption(_name="regularization-epsilon", _prefix=prefix) )
+{
+    this->log("Maxwell","constructor", "start" );
+
+    std::string nameFileConstructor = this->scalabilityPath() + "/" + this->scalabilityFilename() + ".MaxwellConstructor.data";
+    std::string nameFileSolve = this->scalabilityPath() + "/" + this->scalabilityFilename() + ".MaxwellSolve.data";
+    std::string nameFilePostProcessing = this->scalabilityPath() + "/" + this->scalabilityFilename() + ".MaxwellPostProcessing.data";
+    std::string nameFileTimeStepping = this->scalabilityPath() + "/" + this->scalabilityFilename() + ".MaxwellTimeStepping.data";
+    this->addTimerTool("Constructor",nameFileConstructor);
+    this->addTimerTool("Solve",nameFileSolve);
+    this->addTimerTool("PostProcessing",nameFilePostProcessing);
+    this->addTimerTool("TimeStepping",nameFileTimeStepping);
+
+    this->setFilenameSaveInfo( prefixvm(this->prefix(),"Maxwell.info") );
+    //-----------------------------------------------------------------------------//
+    // option in cfg files
+    this->loadParameterFromOptionsVm();
+    //-----------------------------------------------------------------------------//
+    // build mesh
+    if ( buildMesh )
+        this->createMesh();
+    //-----------------------------------------------------------------------------//
+    this->log("Maxwell","constructor", "finish");
+
+}
+
+MAXWELL_CLASS_TEMPLATE_DECLARATIONS
+void
+MAXWELL_CLASS_TEMPLATE_TYPE::loadParameterFromOptionsVm()
+{
+
+}
+
+MAXWELL_CLASS_TEMPLATE_DECLARATIONS
+void
+MAXWELL_CLASS_TEMPLATE_TYPE::createMesh()
+{
+    this->log("Maxwell","createMesh", "start");
+    this->timerTool("Constructor").start();
+
+    createMeshModel<mesh_type>(*this,M_mesh,this->fileNameMeshPath());
```

Comment:
@romainhild I use all the time the .geo in the toolboxes and I have no error. When I use a .geo file, You don't need to put ```gmsh.partition=1``` , because the mesh is already partitioned. I just test with the qs_laplacian and indeed, if you put ```gmsh.partition=1```, there is a crash.

Reply:
ah ok, indeed it makes sense. Anyway, having the check on the empty directory does not hurt.
But can you reproduce the crash in debug ?